Common/include/containers/CLookUpTable.hpp

```diff
@@ -44,48 +44,66 @@ class CLookUpTable {
  protected:
   int rank; /*!< \brief MPI Rank. */
 
-  std::string file_name_lut;
-  std::string version_lut;
-  std::string version_reader;
-  unsigned long n_points;
-  unsigned long n_triangles;
-  unsigned long n_variables;
-  unsigned long n_hull_points;
+  std::string file_name_lut,  /*!< \brief LUT file name.*/ 
+              version_lut,    /*!< \brief LUT version as specified in LUT file.*/ 
+              version_reader, /*!< \brief Reader version (should be equal or above LUT version).*/ 
+              name_CV1,       /*!< \brief Name of controlling variable 1.*/ 
+              name_CV2;       /*!< \brief Name of xontrolling variable 2.*/ 
 
+  su2vector<unsigned long>  n_points,       /*!< \brief Number of data poins per table level.*/ 
+                            n_triangles,    /*!< \brief Number of triangles per table level.*/ 
+                            n_hull_points;  /*!< \brief Number of outer boundary points per table level.*/ 
+
+  unsigned long n_variables,      
+                n_table_levels=1;
+
+  su2vector<su2double> z_values_levels; /*!< \brief Constant z-values of each table level.*/ 
+
+  unsigned short table_dim = 2;/*!< \brief Table dimension.*/ 
   /*! 
-   * \brief the lower and upper limits of the enthalpy and progress variable.
+   * \brief the lower and upper limits of the z, y and x variable for each table level.
    */
-  su2double limits_table_enth[2];
-  su2double limits_table_prog[2];
+  std::pair<su2double, su2double> limits_table_z;
+  su2vector<std::pair<su2double, su2double>> limits_table_y, 
+                                             limits_table_x;
 
   /*! \brief Holds the variable names stored in the table file.
    * Order is in sync with data
    */
-  std::vector<std::string> names_var;
+  su2vector<std::string> names_var;
 
   /*! \brief
    * Holds all data stored in the table. First index addresses the variable
    * while second index addresses the point.
    */
-  su2activematrix table_data;
+  su2vector<su2activematrix> table_data;
 
-  su2matrix<unsigned long> triangles;
+  /*! \brief
+   * Holds all connectivity data stored in the table for each level. First index 
+   * addresses the variable while second index addresses the point.
+   */
+  su2vector<su2matrix<unsigned long>>triangles;
 
-  /* we do not know this size in advance until we go through the entire lookup table */
-  std::vector<std::vector<unsigned long> > edges;
-  std::vector<std::vector<unsigned long> > edge_to_triangle;
+  /*! \brief
+   * Edge information for each table level.
+   */
+  su2vector<std::vector<su2vector<unsigned long>>>edges;
+  su2vector<su2vector<std::vector<unsigned long>>>edge_to_triangle;
```

Comment:
Is it necessary to mix vector types?

Reply:
Unfortunately yes. For 'edges', the outer vector can be an su2vector since the length is the number of table levels. The vector level below that is of inconsistent size, so it has to be an std::vector. The most inner vector is of length 2, so it can be an su2vector. For 'edge_to_triangle', it's similar.

Reply:
If something has a known constant size it should be a std::array which will avoid heap allocation.